dagstore.go

```diff
@@ -333,10 +337,28 @@ func (d *DAGStore) Start(ctx context.Context) error {
 	return nil
 }
 
+func (d *DAGStore) GetIterableIndex(key shard.Key) (carindex.IterableIndex, error) {
+	fi, err := d.indices.GetFullIndex(key)
+	if err != nil {
+		return nil, err
+	}
+
+	ii, ok := fi.(carindex.IterableIndex)
+	if !ok {
+		return nil, errors.New("index for shard is not iterable")
+	}
+
+	return ii, nil
+}
+
 func (d *DAGStore) GetShardKeysForCid(c cid.Cid) ([]shard.Key, error) {
 	return d.invertedIndex.GetShardsForMultihash(c.Hash())
 }
 
+func (d *DAGStore) GetShardKeysForMultihash(h mh.Multihash) ([]shard.Key, error) {
+	return d.invertedIndex.GetShardsForMultihash(h)
+}
+
```

Comment:
Method names are unspecific, as they do not imply what the relationship between the shard and CID/MH is. Please rename to "ShardsContainingCid" and "ShardsContainingMultihash"

(Not sure I like having two variants, either)

Reply:
Changed to having only the `ShardsContainingMultihash ` variant as we can always go from a cid -> multihash.